apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns.php

```diff
@@ -45,35 +69,26 @@ public static function get_instance() {
 	/**
 	 * Register FSE block patterns and categories.
 	 */
-	public function register_patterns() {
-		// Register Block Pattern Categories.
-		if ( class_exists( 'WP_Block_Pattern_Categories_Registry' ) ) {
-			register_block_pattern_category( 'blog', array( 'label' => _x( 'Blog', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'call-to-action', array( 'label' => _x( 'Call to Action', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'contact', array( 'label' => _x( 'Contact', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'images', array( 'label' => _x( 'Images', 'Block pattern category', 'full-site-editing' ) ) );
-			register_block_pattern_category( 'list', array( 'label' => _x( 'List', 'Block pattern category', 'full-site-editing' ) ) );
-
-			// The 'Two Columns of Text' pattern is in the 'columns' and 'text' categories.
-			// Removing 'columns' so it doesn't appear as a category with only a single item.
-			if ( \WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'columns' ) ) {
-				unregister_block_pattern_category( 'columns' );
-			}
-		}
+	private function register_patterns() {
+		$block_patterns = $this->get_patterns();
 
-		if ( class_exists( 'WP_Block_Patterns_Registry' ) ) {
-			// Remove core patterns except 'Two Columns of Text'.
-			// Unfortunately, \WP_Block_Patterns_Registry::get_instance()->get_all_registered() doesn't return the pattern names as keys.
-			foreach ( \WP_Block_Patterns_Registry::get_instance()->get_all_registered() as $pattern ) {
-				if ( 'core/' === substr( $pattern['name'], 0, 5 ) && 'core/text-two-columns' !== $pattern['name'] ) {
-					unregister_block_pattern( $pattern['name'] );
-				}
+		foreach ( (array) $this->get_patterns() as $pattern ) {
+			foreach ( (array) $pattern['tags'] as $slug => $tag ) {
+				register_block_pattern_category( $slug, array( 'label' => $tag['title'] ) );
 			}
 
-			// Add our patterns.
-			foreach ( $this->get_patterns() as $name => $pattern ) {
-				register_block_pattern( $name, $pattern );
-			}
+			register_block_pattern(
+				Block_Patterns::PATTERN_NAMESPACE . $pattern['name'],
+				array(
+					'title'         => $pattern['title'],
+					'description'   => $pattern['title'],
+					'content'       => $pattern['html'],
+					'viewportWidth' => 1280,
+					'categories'    => array_keys(
+						$pattern['tags']
```

Comment:
Just another question, there was some discussion in relation to #42906 (add support for keywords) and in @ramonjd's diff D49335-code about using the pattern's categories for the category here (excluding the category "pattern" of course) so that we can free up the Tags taxonomy to map to keywords instead. More discussion in pbxlJb-nc-p2#comment-464

Is it possible for us to move toward mapping categories to categories? This would require some re-organisation / tagging / categorising of the existing patterns on dotcompatterns, but would be easier to do before this PR lands rather than after 😅

Reply:
@ianstewart I think this is doable, what do you think? It will take a bit of work.

Reply:
That seems OK. Our future selves will thank us when we explain how it works.

- We'd have a set of 10–50 of arbitrary categories shared by Patterns and Page Layouts. The categories would map label for label with the ones in the inserter.
- Some posts would be tagged as Patterns and show up in the Pattern list in the inserter. Some posts would be tagged as Page Layouts and would show up in the Add New Page Screen in the future. Some might be tagged as both and show up in both places.

ToDo

- Ian updates the site to reflect that. :D

Does that sound right?

Reply:
That sounds right to me!

Reply:
Marking this as resolved based on progress in https://github.com/Automattic/view-design/issues/91